tests/std/tests/Dev11_0836436_get_time/test.cpp

```diff
@@ -944,3 +946,265 @@ void test_gh_4820() {
         assert(wss.str() == L"\x043a%\x043e%\x0448%E\x043a%O\x0430");
     }
 }
+
+void test_gh_4882() {
+    // GH-4882 <iomanip>: std::put_time should set failbit on invalid/out-of-range tm struct, instead of crash
+    time_t t = time(nullptr);
+    tm currentTime;
+
+    // Check tm_sec - START
+    localtime_s(&currentTime, &t);
+    currentTime.tm_sec = -1; // set invalid lower limit
+
+    {
+        wstringstream wss;
+        wss << put_time(&currentTime, L"%Y-%m-%d-%H-%M");
+        assert(wss.rdstate() == ios_base::failbit);
+    }

```

Comment:
The following code is highly repetitive. Could we extract this into a helper function, and tell it which data member to damage to what value? That could be done with either a series of lambdas, or a constant table of pointers-to-data-members and ints.

Usually having somewhat repetitive test code is fine (makes it easier to figure out what the test is doing, and debug into it, compared to having to dig through layers of test abstractions), but I think the volume here is worth reducing.